packager/VisibilityChecker.cc

```diff
@@ -360,14 +342,15 @@ class VisibilityCheckerPass final {
         if (!otherPackage.exists() || this->package.mangledName() == otherPackage) {
             return;
         }
+        auto &pkg = ctx.state.packageDB().getPackageInfo(otherPackage);
 
-        bool isExported = false;
+        bool isExported = pkg.exportAll();
         if (litSymbol.isClassOrModule()) {
-            isExported = litSymbol.asClassOrModuleRef().data(ctx)->flags.isExported;
+            isExported |= litSymbol.asClassOrModuleRef().data(ctx)->flags.isExported;
         } else if (litSymbol.isFieldOrStaticField()) {
-            isExported = litSymbol.asFieldRef().data(ctx)->flags.isExported;
+            isExported |= litSymbol.asFieldRef().data(ctx)->flags.isExported;
         }
-        isExported = isExported || db.allowRelaxedPackagerChecksFor(this->package.mangledName());
+        isExported |= db.allowRelaxedPackagerChecksFor(this->package.mangledName());
```

Comment:
These will evaluate the RHS even if `isExported` is already true, right? Should they be changed to `= isExported || ...` to preserve short-circuiting?

Reply:
good catch